src/Param.cc

```diff
@@ -428,7 +430,103 @@ bool ParseColorUsingStringStream(const std::string &_input,
 }
 
 //////////////////////////////////////////////////
-bool Param::ValueFromString(const std::string &_value)
+/// \brief Helper function for Param::ValueFromString for parsing pose
+/// This checks the pose components specified in _input are xyzrpy or xyzwxyz
+/// (expects 6 ir 7 values) and the resulting pose is valid.
+/// \param[in] _input Input string.
+/// \param[in] _key Key of the parameter, used for error message.
+/// \param[in] _attributes Attributes associated to this pose, nullptr if there
+/// are none.
+/// \param[out] _value This will be set with the parsed value.
+/// \return True if parsing pose succeeded.
+bool ParsePoseUsingStringStream(const std::string &_input,
+    const std::string &_key, const Param_V *_attributes,
+    ParamPrivate::ParamVariant &_value)
+{
+  StringStreamClassicLocale ss(_input);
+  std::string token;
+  std::vector<double> values;
+  double v;
+  bool isValidPose = true;
+  while (ss >> token)
+  {
+    try
+    {
+      v = std::stod(token);
+    }
+    // Catch invalid argument exception from std::stod
+    catch(std::invalid_argument &)
+    {
+      sdferr << "Invalid argument. Unable to set value ["<< token
+             << "] for key [" << _key << "].\n";
+      isValidPose = false;
+      break;
+    }
+    // Catch out of range exception from std::stof
+    catch(std::out_of_range &)
+    {
+      sdferr << "Out of range. Unable to set value [" << token
+             << "] for key [" << _key << "].\n";
+      isValidPose = false;
+      break;
+    }
+
+    if (values.size() > 7)
+    {
+      sdferr << "Pose values cannot have more than 7 values.\n";
+      isValidPose = false;
+      break;
+    }
+
+    if (!std::isfinite(v))
+    {
+      sdferr << "Pose values must be finite.\n";
+      isValidPose = false;
+      break;
+    }
+
+    values.push_back(v);
+  }
+
+  if (!isValidPose || !_attributes)
+    return ParseUsingStringStream<ignition::math::Pose3d>(_input, _key, _value);
```

Comment:
Since we've done the work of parsing the string to doubles, how about just constructing a `Pose3d` here instead of calling `ParseUsingStringStream`?

Reply:
`isValidPose` will be set to `false` if any one the tokenized strings is somehow not parse-able to a double, which breaks the parsing loop, https://github.com/ignitionrobotics/sdformat/blob/aaron/pose-configurable-rotation/src/Param.cc#L458-L464

I opted to turn to the previous method of parsing as a last resort on the off chance it might be accepted, and to also be as backward compatible as possible. Let me know what you think.

Reply:
Per VC, let's remove this `if` block so that if there's something not working properly, we'll catch it and fix it instead of allowing it to go through.

Reply:
Removed in https://github.com/ignitionrobotics/sdformat/pull/589/commits/45c243e7abb0efdc0374d94508320a47d5c7b133